### PR TITLE
feat: [CDS-80395]: Org project split view

### PR DESCRIPTION
**Pull request by MondalSayantan**
### Summary

- https://www.figma.com/file/wxzEc30DbiOyJt8dVMFl1f/Left-Nav-2.0?node-id=2106%3A423542&mode=dev
- Added new project-org split view in project selector 

#### Screenshots

NA

https://github.com/harness/harness-core-ui/assets/62999423/bebaa668-93ff-4fca-9b64-164646e28337

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.module.scss [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+.table {
+  padding: var(--spacing-none) var(--spacing-xxlarge);
+}
+
+.colorbox {
```
Comment (avinashmadhwani02) [inline src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.module.scss:12]: Are we using this?

**Review by avinashmadhwani02 — CHANGES_REQUESTED** [review]

File: src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        usePortal={false}
+      >
+        <Icon name="nav-organization" size={24} />
+      </Popover>
+    </Layout.Horizontal>
+  )
+}
+
+interface ProjectScopeSelectorProps {
+  projData: any
```
Comment (avinashmadhwani02) [inline src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx:72]: why any?

Comment (avinashmadhwani02) [reply src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx:72]: avoid use of any?
are you not getting eslint error?

File: src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx [hunk 3]
```diff
@@ -213,8 +233,21 @@ export const ProjectScopeSelector: React.FC<ProjectScopeSelectorProps> = ({ onCl
             </div>
           ) : null}
         </>
-      ) : !loading ? (
-        <NoDataCard icon="nav-project" message={getString('noProjects')} />
+      ) : null}
+      {projectView === Views.SPLIT_VIEW ? (
+        <div className={css.projectContainerWrapper} style={{ display: 'flex', overflow: 'hidden' }}>
+          <OrgProjectSplitView
+            onClick={onClick}
```
[the remark this answers is not in the document]
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx]: variable name changed for more clarity

File: src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx [hunk 4]
```diff
@@ -213,8 +233,21 @@ export const ProjectScopeSelector: React.FC<ProjectScopeSelectorProps> = ({ onCl
...
+            projData={data}
+            projDataLoading={loading}
```
[the remark this answers is not in the document]
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx]: done

File: src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx [hunk 5]
```diff
@@ -65,11 +67,12 @@ interface ProjectScopeSelectorProps {
 export const ProjectScopeSelector: React.FC<ProjectScopeSelectorProps> = ({ onClick }): JSX.Element => {
   const { getString } = useStrings()
   const { accountId } = useParams<OrgPathProps>()
-  const { selectedProject } = useAppStore()
+  const { selectedProject, selectedOrg: org } = useAppStore()
   const { preference: sortPreference = SortMethod.LastModifiedDesc, setPreference: setSortPreference } =
```
[the remark this answers is not in the document]
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx:71]: created new preferencestore as discussed `projectSelectorViewTypeV2`

File: src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+            onClick={() => setSelectedOrg('')}
+            className={cx(css.link, { [css.active]: selectedOrg === '' || selectedOrg === undefined }, css.allButton)}
+          >
+            <Text font={{ variation: FontVariation.BODY }} color={Color.GREY_600} lineClamp={1}>
+              All Organizations
+            </Text>
+          </Container>
+          <div className={css.divider} />
+          {orgData?.data?.content?.length ? (
+            <Layout.Vertical padding={{ right: 'medium' }} height={'82%'} style={{ overflow: 'auto' }}>
```
[the remark this answers is not in the document]
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx]: removed inline styling

File: src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+      }
+    ],
+    []
+  )
+
+  return (
+    <>
+      {(projDataLoading || orgDataLoading) && <PageSpinner />}
+      <Layout.Horizontal style={{ width: '100%' }} margin={{ top: 'xxlarge' }}>
+        <Layout.Vertical width={276}>
```
[the remark this answers is not in the document]
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx:127]: This solution is acceptable as discussed over call

File: src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx (hunk 2 above)
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx]: used correct types

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx [hunk 8]
```diff
@@ -144,78 +159,119 @@ export const ProjectScopeSelector: React.FC<ProjectScopeSelectorProps> = ({ onCl
...
+          selectedSortMethod={sortPreference}
+          sortOptions={[...sortByLastModified, ...sortByCreated, ...sortByName]}
+          onSortMethodChange={option => {
+            setSortPreference(option.value as SortMethod)
+          }}
+          totalCount={data?.data?.totalItems}
+          className={css.listHeader}
+        />
+      )}
+      {orgDataLoading && loading ? (
```
Comment (chetanvekariya-harness) [inline src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx:192]: Is this order change intended ? Earlier it was above ListHeader.

File: src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+          <div className={css.divider} />
+          {orgData?.data?.content?.length ? (
+            <Layout.Vertical padding={{ right: 'medium' }} height={'82%'} className={css.orgDataContainer}>
+              {orgData.data.content.map(org => orgButton(org as OrganizationAggregateDTO))}
+            </Layout.Vertical>
+          ) : (
+            <NoDataCard icon="nav-organization" message={getString('projectsOrgs.noOrganizations')} />
+          )}
+          <div className={css.divider} />
+          <Pagination
```
Comment (chetanvekariya-harness) [inline src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx:168]: Looks like pagination will appear even if there no data. Is it intended ? Does feel like good user experience.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx (hunk 9 above)
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/pages/projects/views/OrgProjectSplitView/OrgProjectSplitView.tsx:168]: Pagination does not appear when there is no data
image

File: src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx (hunk 8 above)
Comment (MondalSayantan) [reply src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx:192]: No specific reason for this order change, just seemed well suited here to me

**Pull request merged**